hack/testing/rsyslog/README.md

```diff
@@ -0,0 +1,224 @@
+
+## Contents of Roles
+### rsyslog
+It contains the framework and data for the configuration files to be deployed.
+
+The basic framework borrowed from debops.rsyslog and adjusted to the RHEL/Fedora specification.
+
+- templates have 2 template files, rsyslog.conf.j2 and rules.conf.j2.  The former is used to generate /etc/rsyslog.conf and the latter is for the other configuration files including mmnormalize rulebase and formatter which will be placed in ```{{rsyslog__config_dir}}``` (default to /etc/rsyslog.d) and its subdirectories.
+
+- tasks/main.yaml contains the sceries of tasks to deploy specified set of configuration files.
+
+If rsyslog__viaq is true, the following tasks are executed.
+```
+TASK [rsyslog : Install/Update required packages] 
+TASK [rsyslog : Create required system group] 
+TASK [rsyslog : Create required system user] 
+TASK [rsyslog : Create a work directory] 
+TASK [rsyslog : Create a temp directory for rsyslog.d backup] 
+TASK [rsyslog : Set backup dir name] 
+TASK [rsyslog : Create a backup dir] 
+TASK [rsyslog : Moving the contents of /etc/rsyslog.d to the backup dir] 
+TASK [rsyslog : create rsyslog viaq subdir] 
+TASK [rsyslog : Update directory and file permissions] 
+TASK [rsyslog : Generate main rsyslog configuration] 
+TASK [rsyslog : Generate viaq configuration files in rsyslog.d] 
+TASK [rsyslog : Generate rsyslog viaq configuration files in rsyslog.d/viaq] 
+```
+If rsyslog__example is true, the following tasks are executed.
+```
+TASK [rsyslog : Install/Update required packages] 
+TASK [rsyslog : Create required system group] 
+TASK [rsyslog : Create required system user] 
+TASK [rsyslog : Create a work directory] 
+TASK [rsyslog : Create a temp directory for rsyslog.d backup] 
+TASK [rsyslog : Set backup dir name] 
+TASK [rsyslog : Create a backup dir] 
+TASK [rsyslog : Moving the contents of /etc/rsyslog.d to the backup dir] 
+TASK [rsyslog : create rsyslog viaq subdir] 
+TASK [rsyslog : Update directory and file permissions] 
+TASK [rsyslog : Generate main rsyslog configuration] 
+TASK [rsyslog : Generate exaple configuration files in rsyslog.d] 
+```
+WARNING: Pre-existing rsyslog.conf and configuration files in /etc/rsyslog.d are moved to the backup directory /tmp/rsyslog.d-XXXXXX.  If the pre-existing files need to be merged with the newly deployed files, you need to do it manually.
+
+-defaults/main.yaml defines variables to switch the deployment paths, variables to specify the locations to deploy and the configurations to be deployed.
+
+Describing how the configuration files are defined to be deployed using the viaq case.
+
+Viaq configuration files are defined in {{rsyslog__viaq_rules}} in defaults/main.yaml.  The set is made from the generic modules{rsyslog__conf_global_options, rsyslog__conf_local_modules, rsyslog__conf_network_modules, rsyslog__conf_network_modules} and viaq specific configurations.  
+
```

Comment:
You have here "rsyslog__conf_network_modules" twice

Reply:
Thank you, @sradco !  Indeed...  It was a copy&paste error. :p  Fixed it.